File: data/readable_format/fortran/file543.f90
module kpoints
    
    implicit none

    ! Declarations
    integer nkpoints
    real, dimension (:, :), allocatable :: special_k
    real, dimension (:, :), allocatable :: special_k_orig
    real, dimension (:), allocatable :: weight_k
    real, dimension (:), allocatable :: weight_k_orig
    real, dimension (:, :), allocatable :: scale_k
    character (len = 40) kptpreference 

end module kpoints

program test_kpoints
    use kpoints
    implicit none

    integer :: i, j
    integer, parameter :: dim1 = 3, dim2 = 2

    ! Initialize nkpoints
    nkpoints = 5
    print *, "nkpoints initialized to: ", nkpoints

    ! Allocate and fill special_k
    allocate(special_k(dim1, dim2))
    do i = 1, dim1
        do j = 1, dim2
            special_k(i, j) = i * j * 0.5
        end do
    end do
    print *, "special_k allocated and filled."
    ! Print special_k values
    do i = 1, dim1
        print *, (special_k(i, j), j = 1, dim2)
    end do

    ! Test character string assignment
    kptpreference = "High symmetry path"
    print *, "kptpreference set to: ", kptpreference

    ! Clean up
    deallocate(special_k)
    print *, "special_k deallocated."

end program test_kpoints

File: data/readable_format/cpp/file543.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cassert>

class KPoints {
public:
    int nkpoints;
    std::vector<std::vector<float>> special_k;
    std::vector<std::vector<float>> special_k_orig;
    std::vector<float> weight_k;
    std::vector<float> weight_k_orig;
    std::vector<std::vector<float>> scale_k;

    std::string kptpreference;

    void allocateSpecialK(int dim1, int dim2) {
        special_k.resize(dim1, std::vector<float>(dim2));
    }

    // Other member functions...
};

void testKPoints() {
    KPoints kp;
    kp.nkpoints = 5;
    assert(kp.nkpoints == 5);

    kp.allocateSpecialK(3, 2);
    assert(kp.special_k.size() == 3);
    assert(kp.special_k[0].size() == 2);

    // Fill in the special_k values
    for (size_t i = 0; i < kp.special_k.size(); ++i) {
        for (size_t j = 0; j < kp.special_k[i].size(); ++j) {
            kp.special_k[i][j] = (i + 1) * (j + 1) * 0.5;
            assert(kp.special_k[i][j] == (i + 1) * (j + 1) * 0.5);
        }
    }

    kp.kptpreference = "High symmetry path";
    assert(kp.kptpreference == "High symmetry path");

    std::cout << "All tests passed." << std::endl;
}

int main() {
    testKPoints();
    return 0;
}